repos/spack_repo/builtin/packages/py_dask_mpi/package.py

```diff
@@ -32,3 +42,17 @@ class PyDaskMpi(PythonPackage):
     # jupyter-server-proxy is not a needed dependency; https://github.com/dask/dask-mpi/pull/102
     # this significantly reduces the dependency tree of py-dask-mpi
     patch("remove-dependency-jupyter-proxy.patch", when="@:2022.4.0")
+
+    # fix for python 3.12+, merged into mainline
+    # https://github.com/dask/dask-mpi/pull/123
+    conflicts("^python@3.12:", when="@:2022")
```

Comment:
```suggestion
    conflicts("^python@3.12:", when="@:2024.4.28")
```
If you want to be more specific

Reply:
I doubt a 2024 will be released at this point, so I'm happy with the bound as is as it's clearly matching what is in the package.py